servicetalk-examples/http/redirects/src/main/java/io/servicetalk/examples/http/redirects/ManualRedirectClient.java

```diff
@@ -0,0 +1,122 @@
+        try (HttpClient secureClient = HttpClients.forSingleAddress("localhost", SECURE_SERVER_PORT)
+                // The custom SSL configuration here is necessary only because this example uses self-signed
+                // certificates. For cases when it's enough to use the local trust store, the trust cert supplier
+                // argument for ClientSslConfigBuilder may be skipped.
+                .sslConfig(new ClientSslConfigBuilder(DefaultTestCerts::loadServerCAPem).build()).build()) {
+
+            try (HttpClient client = HttpClients.forSingleAddress("localhost", NON_SECURE_SERVER_PORT).build()) {
+                // This example is demonstrating asynchronous execution, but needs to prevent the main thread from exiting
+                // before the response has been processed. This isn't typical usage for a streaming API but is useful for
+                // demonstration purposes.
+                CountDownLatch responseProcessedLatch = new CountDownLatch(1);
+
+                // Redirect of a GET request with a custom header:
+                HttpRequest originalGet = client.get("http://localhost:8080/sayHello")
+                        .addHeader(CUSTOM_HEADER, "value");
+                client.request(originalGet)
+                        .flatMap(response -> {
+                            if (response.status().statusClass() == REDIRECTION_3XX) {
+                                CharSequence location = response.headers().get(LOCATION);
+                                HttpClient redirectClient = lookupClient(location, client, secureClient);
+                                return redirectClient.request(redirectClient
+                                        .newRequest(originalGet.method(), location.toString())
+                                        .addHeader(CUSTOM_HEADER, originalGet.headers().get(CUSTOM_HEADER)));
+                            }
+                            // Decided not to follow redirect, return the original response or an error:
+                            return succeeded(response);
+                        })
+                        .afterFinally(responseProcessedLatch::countDown)
+                        .subscribe(resp -> {
+                            System.out.println(resp.toString((name, value) -> value));
+                            System.out.println(resp.payloadBody(textDeserializer()));
+                            System.out.println();
+                        });
+
+                responseProcessedLatch.await();
+
+                // Redirect of a POST request with a payload body:
+                responseProcessedLatch = new CountDownLatch(1);
+                HttpRequest originalPost = client.post("http://localhost:8080/sayHello")
+                        .payloadBody(client.executionContext().bufferAllocator().fromAscii("some_content"));
+                client.request(originalPost)
+                        .flatMap(response -> {
```

Comment:
Handling for multiple redirects?

Reply:
It's possible, but in the case of a manual redirect users have to take care of it. By playing around, users can write a method that will apply this `flatMap` operator, then re-use the method recursively from inside `flatMap`. It's also possible to pass a counter there, which will be used to limit the number of recursive calls. I will leave all these enhancements to the users' decision. Otherwise, the scope of this example may grow.